src/Testcontainers/Configurations/WaitStrategies/UntilHttpRequestIsCompleted.cs

```diff
@@ -0,0 +1,94 @@
+namespace DotNet.Testcontainers.Configurations
+{
+  using System;
+  using System.Linq;
+  using System.Net.Http;
+  using System.Threading;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Builders;
+  using DotNet.Testcontainers.Containers;
+  using Microsoft.Extensions.Logging;
+
+  internal class UntilHttpRequestIsCompleted : IWaitUntil, IDisposable
+  {
+    private readonly HttpWaitRequest request;
+    private readonly int frequency;
+    private readonly int timeout;
+    private readonly HttpClientHandler handler;
+    private readonly bool disposeHandler;
+
+    public UntilHttpRequestIsCompleted(HttpWaitRequest request, int frequency, int timeout)
+      : this(request, frequency, timeout, new HttpClientHandler(), true)
+    {
+    }
+
+    public UntilHttpRequestIsCompleted(HttpWaitRequest request, int frequency, int timeout, HttpClientHandler handler, bool disposeHandler = false)
+    {
+      this.request = request;
+      this.handler = handler;
+      this.disposeHandler = disposeHandler;
+      this.frequency = frequency;
+      this.timeout = timeout;
+    }
+
+    public async Task<bool> Until(ITestcontainersContainer testcontainers, ILogger logger)
+    {
+      var httpClient = new HttpClient(this.handler, false);
+      httpClient.Timeout = this.request.ReadTimeout;
+
+      await WaitStrategy.WaitUntil(
+        async () =>
+        {
+          try
+          {
+            var response = await httpClient.SendAsync(
+              new HttpRequestMessage(
+                this.request.Method,
+                this.BuildRequestUri(testcontainers.Hostname, testcontainers.GetMappedPublicPort(this.request.Port))));
```

Comment:
Maybe you can pass the `Hostname` to something like `HttpRequestMessage Build(string hostname, int mappedPublicPort)`. That the builder creates the `HttpRequestMessage`. Then you probably don't need the nested `Builder` class too.